backend/Events/NightTimeVoteEvents.js

```diff
@@ -0,0 +1,55 @@
+const RoleEnum = require('../domain/Enum/Role');
+const SuspectRevealDTO = require('../domain/DTO/response/SuspectRevealDTO');
+
+/**
+ * Event handlers and logic for `mafia-vote`, 'medic-vote' and `detective-vote`
+ * This involves the functionality for night-time voting, which includes:
+ * - Mafia choosing who to kill before the next round
+ * - Medics choosing who to save
+ * - Detectives choosing who to suspect
+ * @param {any} io
+ * @param {any} socket
+ * @param {MafiaGame} mafiaGame
+ */
+exports.loadNightTimeEvents = (io, socket, mafiaGame) => {
+    /**
+     * Handler for 'mafia-vote', sets the chosen player for the Mafia to the player specified in the message.
+     * In future iterations, this will be changed to tally votes from mafia members before setting the chosen player.
+     */
+    socket.on('mafia-vote', (mafiaVoteObj) => {
+        let room = mafiaGame.gameRoomsDict[socket.player.roomID];
+        room.voteHandler.addMafiaVote(
+            socket.player,
+            room.getPlayerByNickname(mafiaVoteObj.votingFor)
+        );
+    });
+
+    /**
+     * Handler for 'medic-vote', pretty much the same logic as the mafia vote, except it sets the chosen player for the Medics.
+     */
+    socket.on('medic-vote', (medicVoteObj) => {
```

Comment:
Can you also create the request DTOs?

Reply:
I have added a NightTimeVoteDTO, which can be used for all of these events, since they carry the same info